src/renderer/components/message/Message.tsx

```diff
@@ -88,7 +88,19 @@ const AuthorName = (
   onContactClick: (contact: T.Contact) => void,
   overrideSenderName?: string
 ) => {
-  const { color, displayName } = contact
+  const { id, color } = contact
+  const _displayName = contact.displayName
+  const [displayName, setDisplayName] = useState<string>(_displayName)
+  const update = async ({ contactId }: { contactId: number | null }) => {
+    if (contactId === id) {
+      const contact = await BackendRemote.rpc.getContact(
+        selectedAccountId(),
+        contactId
+      )
+      setDisplayName(contact.displayName)
+    }
+  }
+  onDCEvent(selectedAccountId(), 'ContactsChanged', update)
```

Comment:
you need to wrap this in useEffect, `onDCEvent` returns a cleanup function that unregisters the even listener, useEffect makes sure it is called.

See the other places where `onDCEvent` is used